src/test/java/com/google/sps/unit/ServletUtilsTest.java

```diff
@@ -0,0 +1,113 @@
+    when(httpServletRequest.getParameter("myKey")).thenReturn("myValue");
+
+    Optional<String> parameter =
+        ServletUtils.getRequestParam(httpServletRequest, httpServletResponse, "myKey");
+    Assert.assertTrue(parameter.isPresent());
+    Assert.assertEquals(parameter.get(), "myValue");
+  }
+
+  // Test ServletUtils getRquestParam() function correctly returns empty Optional container when
+  // parameter key does not exist.
+  @Test
+  public void testHTTPRequestGetInvalidParameter() throws Exception {
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    Optional<String> parameter =
+        ServletUtils.getRequestParam(httpServletRequest, httpServletResponse, "anInvalidKey");
+    Assert.assertFalse(parameter.isPresent());
+    verify(printWriter).println("No anInvalidKey in the query URL.");
+  }
+
+  // Test ServletUtils getRquestParam() function correctly returns empty Optional container when
+  // provided parameter key is null.
+  @Test
+  public void testHTTPRequestGetNullParameter() throws Exception {
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    Optional<String> parameter =
+        ServletUtils.getRequestParam(httpServletRequest, httpServletResponse, null);
+    Assert.assertFalse(parameter.isPresent());
+    verify(printWriter).println("No null in the query URL.");
+  }
+
+  // Test finding an Election Entity in Datastore
+  @Test
+  public void testFindExistingElectionInDatastore() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    Entity electionEntity = new Entity("Election");
+    electionEntity.setProperty("id", "9999");
+    electionEntity.setProperty("name", "myElection");
+    electionEntity.setProperty("scope", "myScope");
+    electionEntity.setProperty("date", "myDate");
+    electionEntity.setProperty("contests", new HashSet<Long>());
+    electionEntity.setProperty("propositions", new HashSet<Long>());
+    ds.put(electionEntity);
+
+    Optional<Entity> foundEntity = ServletUtils.findElectionInDatastore(ds, "9999");
+    Assert.assertTrue(foundEntity.isPresent());
+    Assert.assertEquals(foundEntity.get(), electionEntity);
+  }
+
+  // Test finding an Election Entity that does not exist in Datastore
+  @Test
+  public void testFindNonexistingElectionInDatastore() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    Entity electionEntity = new Entity("Election");
+    electionEntity.setProperty("id", "9999");
+    electionEntity.setProperty("name", "myElection");
+    electionEntity.setProperty("scope", "myScope");
+    electionEntity.setProperty("date", "myDate");
+    electionEntity.setProperty("contests", new HashSet<Long>());
+    electionEntity.setProperty("propositions", new HashSet<Long>());
+    ds.put(electionEntity);
+
+    Optional<Entity> foundEntity = ServletUtils.findElectionInDatastore(ds, "0001");
+    Assert.assertFalse(foundEntity.isPresent());
```

Comment:
(This is FYI only, no action required):

I would recommend using Truth for fluent assertions if this was google internal code:
https://truth.dev/

Which generally makes assertions more readable. But we can skip it for now since it's a bit more setup for google external.